x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -296,26 +296,48 @@ func (k Keeper) updateSupplier(
 	currentHeight := sdkCtx.BlockHeight()
 	nextSessionStartHeight := sharedtypes.GetNextSessionStartHeight(&sharedParams, currentHeight)
 
-	// If new service configs were provided, mark all the supplier's service
-	// configurations as to be deactivated at the start of the next session.
-	if len(msg.Services) > 0 {
-		for _, oldServiceConfigUpdate := range supplier.ServiceConfigHistory {
-			oldServiceConfigUpdate.DeactivationHeight = nextSessionStartHeight
-		}
-	}
+	updatedServiceConfigHistory := make([]*sharedtypes.ServiceConfigUpdate, 0)
+	updatedServices := make(map[string]struct{})
 
-	// Initialize the supplier's service configurations with the new ones from the msg.
-	// These will take effect at the start of the next session.
+	// Step 1: Add all new service configurations from the message
+	// - These configs will activate at the start of the next session
+	// - Track service IDs to identify which old inactive configs need replacement
 	for _, newServiceConfig := range msg.Services {
 		newServiceConfigUpdate := &sharedtypes.ServiceConfigUpdate{
 			OperatorAddress: msg.OperatorAddress,
 			Service:         newServiceConfig,
 			// The effective block height is the start of the next session.
 			ActivationHeight: nextSessionStartHeight,
 		}
-		supplier.ServiceConfigHistory = append(supplier.ServiceConfigHistory, newServiceConfigUpdate)
+		updatedServiceConfigHistory = append(updatedServiceConfigHistory, newServiceConfigUpdate)
+		updatedServices[newServiceConfig.ServiceId] = struct{}{}
+	}
+
+	// Step 2: Handle existing service configurations
+	if len(msg.Services) > 0 {
+		for _, oldServiceConfigUpdate := range supplier.ServiceConfigHistory {
```

Comment:
I'm getting really confused by the terminology:
- old
- update
- new
- etc...

Is there any way can consolidate it into two?

E.g.
- currentServiceConfigs
- newServiceConfigs